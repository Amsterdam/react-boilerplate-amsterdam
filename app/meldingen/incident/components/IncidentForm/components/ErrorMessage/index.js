import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = ({ touched, hasError }) => (
  <div className="field-errors">
    <div>
      {touched
      && hasError('required')
      && 'Dit is verplicht veld'}
    </div>

    <div>
      {touched
      && hasError('email')
      && 'Het moet een geldig e-mailadres zijn'}
    </div>
  </div>
);

ErrorMessage.propTypes = {
  touched: PropTypes.bool.isRequired,
  hasError: PropTypes.func.isRequired
};

export default ErrorMessage;
