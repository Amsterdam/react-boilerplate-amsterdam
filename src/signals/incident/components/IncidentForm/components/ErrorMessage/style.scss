@import "config.scss";

.field-errors {
  &__item {
    color: $ams-rood;
    font-size: 14px;
    margin-bottom: 4px;
  }
}
