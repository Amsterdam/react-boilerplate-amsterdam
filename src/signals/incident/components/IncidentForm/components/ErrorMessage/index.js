import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';

const ErrorMessage = ({ touched, hasError, getError }) => (
  <div className="field-errors">
    <div className="field-errors__item">
      {touched
      && hasError('required')
      && 'Dit is een verplicht veld'}
    </div>

    <div className="field-errors__item">
      {touched
      && hasError('email')
      && 'Het moet een geldig e-mailadres zijn'}
    </div>

    <div className="field-errors__item">
      {touched
      && hasError('maxLength')
      && `Dit veld mag niet langer dan ${getError('maxLength').requiredLength} characters zijn.`}
    </div>
  </div>
);

ErrorMessage.propTypes = {
  touched: PropTypes.bool.isRequired,
  hasError: PropTypes.func.isRequired,
  getError: PropTypes.func.isRequired
};

export default ErrorMessage;
