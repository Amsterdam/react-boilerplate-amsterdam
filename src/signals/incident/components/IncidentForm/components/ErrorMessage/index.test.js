import React from 'react';
import { shallow } from 'enzyme';

import ErrorMessage from './index';

describe('Form component <ErrorMessage />', () => {
  let wrapper;
  let touched;
  let getError;
  let hasError;

  beforeEach(() => {
    touched = false;
    getError = jest.fn();
    hasError = jest.fn();

    wrapper = shallow(<ErrorMessage
      touched={touched}
      hasError={hasError}
      getError={getError}
    />);
  });

  describe('rendering', () => {
    it('should render no error by default', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render no error when element is only touched', () => {
      wrapper.setProps({ touched: true });

      expect(wrapper).toMatchSnapshot();
    });

    it('should render required error', () => {
      hasError.mockImplementation((type) => type === 'required');
      wrapper.setProps({ touched: true });

      expect(wrapper).toMatchSnapshot();
    });

    it('should render email error', () => {
      hasError.mockImplementation((type) => type === 'email');
      wrapper.setProps({ touched: true });

      expect(wrapper).toMatchSnapshot();
    });

    it('should render maxLength error', () => {
      hasError.mockImplementation((type) => type === 'maxLength');
      getError.mockImplementation(() => ({
        requiredLength: 666
      }));
      wrapper.setProps({ touched: true });

      expect(wrapper).toMatchSnapshot();
    });
  });
});
