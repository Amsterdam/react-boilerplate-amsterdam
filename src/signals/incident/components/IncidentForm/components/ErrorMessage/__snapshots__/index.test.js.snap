// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component <ErrorMessage /> rendering should render email error 1`] = `
<div
  className="field-errors"
>
  <div
    className="field-errors__item"
  />
  <div
    className="field-errors__item"
  >
    Het moet een geldig e-mailadres zijn
  </div>
  <div
    className="field-errors__item"
  />
</div>
`;

exports[`Form component <ErrorMessage /> rendering should render maxLength error 1`] = `
<div
  className="field-errors"
>
  <div
    className="field-errors__item"
  />
  <div
    className="field-errors__item"
  />
  <div
    className="field-errors__item"
  >
    Dit veld mag niet langer dan 666 characters zijn.
  </div>
</div>
`;

exports[`Form component <ErrorMessage /> rendering should render no error by default 1`] = `
<div
  className="field-errors"
>
  <div
    className="field-errors__item"
  />
  <div
    className="field-errors__item"
  />
  <div
    className="field-errors__item"
  />
</div>
`;

exports[`Form component <ErrorMessage /> rendering should render no error when element is only touched 1`] = `
<div
  className="field-errors"
>
  <div
    className="field-errors__item"
  />
  <div
    className="field-errors__item"
  />
  <div
    className="field-errors__item"
  />
</div>
`;

exports[`Form component <ErrorMessage /> rendering should render required error 1`] = `
<div
  className="field-errors"
>
  <div
    className="field-errors__item"
  >
    Dit is een verplicht veld
  </div>
  <div
    className="field-errors__item"
  />
  <div
    className="field-errors__item"
  />
</div>
`;
